packages/babel-plugin-transform-modules-amd/src/index.ts

```diff
@@ -12,33 +13,35 @@ import {
 import { template, types as t } from "@babel/core";
 import { getImportSource } from "babel-plugin-dynamic-import-node/utils";
 import type { PluginOptions } from "@babel/helper-module-transforms";
+import type { NodePath } from "@babel/traverse";
 
-const buildWrapper = template(`
+const buildWrapper = template.statement(`
   define(MODULE_NAME, AMD_ARGUMENTS, function(IMPORT_NAMES) {
   })
 `);
 
-const buildAnonymousWrapper = template(`
+const buildAnonymousWrapper = template.statement(`
   define(["require"], function(REQUIRE) {
   })
 `);
 
-function injectWrapper(path, wrapper) {
+function injectWrapper(path: NodePath<t.Program>, wrapper: t.Statement) {
   const { body, directives } = path.node;
   path.node.directives = [];
   path.node.body = [];
   const amdWrapper = path.pushContainer("body", wrapper)[0];
-  const amdFactory = amdWrapper
-    .get("expression.arguments")
+  const amdFactory = (
+    amdWrapper.get("expression.arguments") as NodePath<t.Expression>[]
+  )
```

Comment:
We should annotate the `wrapper` parameter as `t.ExpressionStatement` (so that this type cast is guaranteed to be correct).

Also, it would probably be better to do `const amdWrapper: NodePath<t.ExpressionStatement> = path.pushContainer("body", wrapper)[0]` instead of type-casting here: it feels safer because we are making less assumptions about the code (we are just helping it to know that `pushContainer` returns its argument wrapped).